pkg/cmd/bazci/bazci.go

```diff
@@ -85,6 +275,12 @@ func init() {
 		false,
 		"process failures artifacts (and post github issues for release failures)",
 	)
+	rootCmd.Flags().IntVar(
+		&port,
+		"port",
+		8998,
+		"port to run the bazci server on",
+	)
 }
 
 func bazciImpl(cmd *cobra.Command, args []string) (retErr error) {
```

Comment:
We don't want to return directly here, if an error should be returned assign it to `retErr` then `return` so the deferred function can run as expected. Previously we would create the tarball if bazci exited with a non-zero exit code, so that's how this code ended like this (after merging #87938).